### PR TITLE
supporting https

### LINKED ISSUE #28: https support?
Needed or not? This tool is obviously meant to be used for systems in development, so not supporting https makes perfect sense, but we need to document it.

### PR DESCRIPTION
**Pull request by thekemkid**
fixes #28

### CODE REVIEW

File: lib/myhttp.js [hunk 1]
```diff
@@ -85,7 +87,12 @@ function Client (opts) {
 inherits(Client, EE)
 
 Client.prototype._connect = function () {
-  this.conn = net.connect(this.opts.port, this.opts.hostname)
+  if (this.secure) {
+    this.conn = tls.connect(this.opts.port, this.opts.hostname, { rejectUnauthorized: false })
```
Comment (thekemkid) [inline lib/myhttp.js]: Should we allow users to specify rejectUnauthorized?

Comment (thekemkid) [inline lib/myhttp.js]: relevant, CTRL-F rejectUnauthorized: https://nodejs.org/api/https.html#https_https_createserver_options_requestlistener

Comment (mcollina) [inline lib/myhttp.js]: I think for a load tester it's good to not validate certificates.

**Comment by mcollina** [comment]
Can you please check the performance of our ssl impl with the one of wrk?

File: README.md [hunk 2]
```diff
@@ -4,8 +4,8 @@
 
 ![demo](https://raw.githubusercontent.com/mcollina/autocannon/master/demo.gif)
 
-An HTTP benchmarking tool written in node, greatly inspired by
-[wrk][wrk] and [wrk2][wrk2], with support to HTTP pipelining.
+A HTTP benchmarking tool written in node, greatly inspired by [wrk][wrk]
```
Comment (mcollina) [inline README.md]: we should probably say that this is a HTTP/1.1

Comment (thekemkid) [inline README.md]: done

**Comment by thekemkid** [comment]
```
Glens-MBP-2:os glen$ ./autocannon/autocannon.js -c 1000 -d 30 https://localhost:3000
Running 30s test @ https://localhost:3000
1000 connections

Stat         Avg     Stdev     Max
Latency (ms) 73.46   89.64     2911
Req/Sec      13609.3 3602.58   15591
Bytes/Sec    1.52 MB 403.83 kB 1.77 MB

408k requests in 31s, 45.73 MB read
28 errors (0 timeouts)
Glens-MBP-2:os glen$ ./wrk/wrk -c 1000 -d 30 https://localhost:3000
Running 30s test @ https://localhost:3000
  2 threads and 1000 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     6.76ms    6.86ms 119.26ms   95.55%
    Req/Sec     5.64k     2.72k    9.08k    60.66%
  309316 requests in 30.03s, 33.04MB read
  Socket errors: connect 18898, read 16, write 0, timeout 80
Requests/sec:  10301.21
Transfer/sec:      1.10MB
```

**Comment by mcollina** [comment]
Can you please check what's going on with the high latency?

**Pull request merged**